src/libraries/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -831,21 +831,10 @@ public sealed override bool Equals(MethodInfo? left, MethodInfo? right)
                     else if (right == null)
                         return false;
 
-                    // TODO: switch to use MemberInfo.MetadataToken here.
-                    // It compares honestly referring ECMA-335 I.8.6.1.6 Signature Matching.
-                    if (!Equals(left.DeclaringType, right.DeclaringType))
+                    if (left.Module.Name != right.Module.Name)
                         return false;
 
-                    if (!Equals(left.ReturnType, right.ReturnType))
-                        return false;
-
-                    if (left.CallingConvention != right.CallingConvention)
-                        return false;
-
-                    if (left.IsStatic != right.IsStatic)
-                        return false;
-
-                    if (left.Name != right.Name)
+                    if (left.MetadataToken != right.MetadataToken)
```

Comment:
Is it possible that left is TypeRef and right is TypeDef? that could potentially produce false where it shouldn't.

Also if the token is equal should it just return true? Could arguments differ for same token?

Reply:
As far as I know, MetadataToken is unique in the same Module, so when MetadataToken is not equal, it can return false. In addition, according to [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.metadatatoken?view=net-5.0&viewFallbackFrom=netstandard-1.0#System_Reflection_MemberInfo_MetadataToken), I am not sure whether it is necessary to judge the unexpected situation. Do you have any suggestions? I will change the code later.